scripts/upload/add_alaska_pits_2023.py

```diff
@@ -1,114 +1,75 @@
     ignore_files = [
-        "SnowEx23_SnowPits_AKIOP_Summary_Environment_v01.csv",
-        "SnowEx23_SnowPits_AKIOP_Summary_SWE_v01.csv",
-        "SnowEx23_SnowPits_AKIOP_Summary_SWE_v01_modified.csv"
+        "Summary_Environment",
+        "Summary_SWE",
+        "Summary_SWE",
     ]
 
-    # Get all the date folders
-    unique_folders = Path(
-        data_dir
-    ).expanduser().absolute().glob("ALASKA*/*20*SNOW_PIT")
-    for udf in unique_folders:
-        # get all the csvs in the folder
-        dt_folder_files = list(udf.glob("*.csv"))
-        site_ids = []
-        # Get the unique site ids for this date folder
-        compiled = re.compile(
-            r'SnowEx23_SnowPits_AKIOP_([a-zA-Z0-9]*)_\d{8}.*_v01\.csv'
-        )
-        for file_path in dt_folder_files:
-            file_name = file_path.name
-            if file_name in ignore_files:
-                print(f"Skipping {file_name}")
-                continue
-            match = compiled.match(file_name)
-            if match:
-                code = match.group(1)
-                site_ids.append(code)
-            else:
-                raise RuntimeError(f"No site ID found for {file_name}")
-
-        # Get the unique site ids
-        site_ids = list(set(site_ids))
-
-        for site_id in site_ids:
-            # Grab all the csvs in the pits folder
-            filenames = glob.glob(join(str(udf), f'*_{site_id}_*.csv'))
-
-            # Grab all the site details files
-            sites = glob.glob(join(
-                str(udf), f'*_{site_id}_*siteDetails*.csv'
-            ))
-
-            # Use no-gap-filled density
-            density_files = glob.glob(join(
-                str(udf), f'*_{site_id}_*_gapFilled_density*.csv'
-            ))
-
-            # Remove the site details from the total file list to get only the
-            profiles = list(
-                set(filenames) - set(sites) -
-                set(density_files)  # remove non-gap-filled denisty
-            )
+    # Data to skip for now
+    ignore_data = [
+        "gapFilled_density",
+    ]
 
-            # Submit all profiles associated with pit at a time
-            b = UploadProfileBatch(
-                filenames=profiles, debug=debug, doi=doi,
-                in_timezone=timezone,
-                db_name=db_name,
-                allow_split_lines=True,  # Logic for split header lines
-                header_sep=":"
-            )
-            b.push()
-            error_msg += b.errors
-
-            # Upload the site details
-            sd = UploadSiteDetailsBatch(
-                filenames=sites, debug=debug, doi=doi,
-                in_timezone=timezone,
-                db_name=db_name
-            )
-            sd.push()
-            error_msg += sd.errors
+    # Filter to CSV and relevant data
+    file_list = [
+        file
+        for file in file_list
+        if str(file).lower().endswith(".csv")
+        and not any(name in file for name in ignore_files)
```

Comment:
For me this needed to be: `name in str(file) for name in ignore_files` and similarly for the next line. I think this depends on local versus downloaded files? I think earthaccess returns Path objects and needs to be converted.

Reply:
I am surprised to see that issue since the list should always be strings.
Here the [line that converts local file paths](https://github.com/SnowEx/snowex_db/blob/5222f9386273e5aefdf54f9a42b2059d521fa6b0/scripts/upload/earthaccess_data.py#L28) and earthaccess always returns list of files as strings.

Reply:
My apologies. I think I misinterpreted the very long load times from earthaccess as a hang in the code, interrupted and then went down a strange rabbit hole. This works fine now.